lisa/operating_system.py

```diff
@@ -261,8 +451,59 @@ def _initialize_package_installation(self) -> None:
                 timeout=3600,
             )
 
-    def _install_packages(self, packages: Union[List[str]]) -> None:
-        self._node.execute(f"yum install -y {' '.join(packages)}", sudo=True)
+    def _install_packages(
+        self, packages: Union[List[str]], signed: bool = True
+    ) -> None:
+        command = f"yum install -y {'|'.join(packages)}"
+        if not signed:
+            command.__add__(" --nogpgcheck")
+
+        install_result = self._node.execute(command, sudo=True)
+        # yum returns exit_code=1 if package is already installed.
+        # We do not want to fail if exit_code=1.
+        if install_result.exit_code == 1:
+            self._log.info(
+                f"{' '.join(packages)} is/are already installed."
+                f" stdout: {install_result.stdout}"
+            )
+        elif install_result.exit_code != 0:
+            raise LisaException(
+                f"Failed to install {' '.join(packages)}."
+                f" stdout: {install_result.stdout}"
+            )
+        else:
+            self._log.info(
+                f"{' '.join(packages)} is/are installed successfully."
+                f" stdout: {install_result.stdout}"
+            )
+
+    def _query_package(self, package: str, signed: bool = True) -> bool:
+        command = f"yum list installed | grep -Ei {package}"
+        result = self._node.execute(command, sudo=True)
+        if result.exit_code == 0:
+            return True
+
+        return False
+
+    def _get_os_version(self) -> OsVersion:
+        cmd_result = self._node.execute(
+            cmd="cat /etc/redhat-release", no_error_log=True
+        )
+        if cmd_result.exit_code == 0 and cmd_result.stdout != "":
+            result = cmd_result.stdout
+            for vendor in ["Red Hat", "CentOS", "Fedora", "XenServer"]:
```

Comment:
What happens if it's not in this list? Raise an exception for the `for/else`  block.

Reply:
My understanding is, if the name is not in this list, it's an unknown situation. We should use guard statements to make those unknown things surfacing out, either fix them or workaround them. After we have the os version feature, it will be used a lot in test cases. If it's empty, it may make unexpected results, or missing some checks.

Reply:
Alright!